test/crash_test.dart

```diff
@@ -50,79 +73,158 @@ void main() async {
           ((await getJson('https://pub.dev/api/package-names'))['packages']
                   as List)
               .cast<String>();
-      print('Found ${packages.length} packages to scan');
+      //.take(3).toList(); // useful when testing
+      print('## Found ${packages.length} packages to scan');
 
-      final errors = <String>[];
-      final pool = Pool(50);
       var count = 0;
-      var skipped = 0;
-      var lastStatus = DateTime.now();
+      final pool = Pool(50);
+      final packageVersions = <PackageVersion>[];
       await Future.wait(packages.map((package) async {
         await pool.withResource(() async {
-          final versionsResponse =
-              await getJson('https://pub.dev/api/packages/$package');
-          final archiveUrl = Uri.tryParse(
-            versionsResponse['latest']?['archive_url'] as String? ?? '',
+          final response = await getJson(
+            'https://pub.dev/api/packages/$package',
           );
+          final entry = response['latest'] as Map?;
+          if (entry != null) {
+            packageVersions.add(PackageVersion(
+              package: package,
+              version: entry['version'] as String,
+              archiveUrl: entry['archive_url'] as String,
+            ));
+          }
+
+          // To analyze all package versions use the following lines.
+          // beware that this extremely, and probably scans that same files
+          // many times for no good reason.
+          /*packageVersions.addAll(
```

Comment:
Commented out lines should not be checked in; they will bit rot, and no one will how to recreate the intenion months after they become dead.

Instead, they can be wrapped with something like `if (shouldAddAllVersions)` and a flag above, `const shouldAddAllVersions = false;`

Reply:
Let's just drop it, it'll take 10x longer, and if we really wanted a larger corpus of markdown files we are better of finding different ones.